controller/main.go

```diff
@@ -46,8 +46,8 @@ type controller struct {
 }
 
 func (c *controller) SetBalancer(l log.Logger, name string, svcRo *v1.Service, _ *v1.Endpoints) k8s.SyncState {
-	l.Log("event", "startUpdate", "msg", "start of service update")
-	defer l.Log("event", "endUpdate", "msg", "end of service update")
+	l.Log("event", "startUpdate", "msg", "start of service update")   //nolint:errcheck
```

Comment:
I'm not sure why a logger should return an error. Changing this is out of scope for this PR, however using `nolint` on every log statement from now on doesn't seem reasonable to me. OTOH I also don't want to disable this linter globally since it can save some nasty debugging in certain cases.

We have https://github.com/metallb/metallb/issues/664 for handling the logging part. I'm not sure whether to move forward with all these `nolint` statements or disable this linter globally as a workaround and add a TODO in the code until the logging task is done.

Reply:
There seems to be a way to exclude certain function calls from certain linters. However, I can't get the syntax to work in the short time I spent on it. I will look again later.

Reply:
I figured it out by downloading errcheck. Going to push a new PR.